sqlglot/dialects/clickhouse.py

```diff
@@ -84,6 +84,39 @@ class Parser(parser.Parser):
 
         LOG_DEFAULTS_TO_LN = True
 
+        def _parse_placeholder(self) -> t.Optional[exp.Expression]:
+            """
+            Parse a placeholder expression like SELECT {abc: UInt32} or FROM {table: Identifier}
+            https://clickhouse.com/docs/en/sql-reference/syntax#defining-and-using-query-parameters
+            """
+            index = self._index
+
+            if self._match(TokenType.L_BRACE):
```

Comment:
1. Can / should we just return `None` if this doesn't match?
2. Can we remove the retreat by returning early if the first two tokens are not `{` and a `VAR` or `TABLE`? Not sure if there are other tokens, but maybe we can get rid of the retreat below.